api/v1beta1/types.go

```diff
@@ -422,18 +422,21 @@ const (
 )
 
 // IdentityType represents different types of identities.
-// +kubebuilder:validation:Enum=ServicePrincipal;ManualServicePrincipal;UserAssignedMSI
+// +kubebuilder:validation:Enum=ServicePrincipal;ManualServicePrincipal;ServicePrincipalCertificate
```

Comment:
this looks like a breaking change, is that intended/allowed?

Reply:
It technically breaking but in practice, no one should be broken by the change. Right now if you set it to `UserAssignedMSI` it won't work and just throw a [validation error in the controller](https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2258/files#diff-b6ab207506064afcbd1d777c1c7296e2953d2653a8843bfb39107af5cf5b0a87L90). It has never been supported and there is an [issue](https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1104) to add support for it (planning on working on that next). I could leave it as-is but IMO this is an improvement as it's better to fail the user right away if they try to use an unsupported value than to let them create the resource and later fail in the controller.

Reply:
Makes sense!